public/apps/configuration/utils/request-utils.ts

```diff
@@ -13,29 +13,85 @@
  *   permissions and limitations under the License.
  */
 
-import { HttpStart, HttpHandler } from 'opensearch-dashboards/public';
+import { HttpStart, HttpHandler, HttpFetchQuery } from 'opensearch-dashboards/public';
 
-export async function request<T>(requestFunc: HttpHandler, url: string, body?: object): Promise<T> {
-  if (body) {
-    return (await requestFunc(url, { body: JSON.stringify(body) })) as T;
-  }
-  return (await requestFunc(url)) as T;
+interface RequestType {
+  http: HttpStart;
+  url: string;
+  body?: object;
+}
+
+interface RequestParams {
```

Comment:
Here is a suggestion to re-structure and re-organize the 4 interfaces introduced in this file:

```ts
// Base interface for common properties of all request types
interface BaseRequest {
  url: string;
  body?: object;
}

// Parent interface for any request type that should ignore specified error codes
interface IgnorableRequest {
  ignores: number[];
}

// Interface for request initiation
interface HttpRequestInit extends BaseRequest {
  httpStart: HttpStart; // starting point of HTTP request
}

// Interface for parameters needed to make a request
interface HttpRequestParams extends BaseRequest {
  requestHandler: HttpHandler; // the request handler
  query: HttpFetchQuery;
}



// Ignorable HttpRequestInit
interface IgnorableHttpRequestInit extends HttpRequestInit, IgnorableRequest {}

// Ignorable HttpRequestParams
interface IgnorableHttpRequestParams extends HttpRequestParams, IgnorableRequest {}

```

Reply:
Addressed in: [c048a43](https://github.com/opensearch-project/security-dashboards-plugin/pull/1888/commits/c048a430440bf2131ba0eb19ce63e0df39eb12a5). I think it makes it more clear, even though there are more interfaces. If there are subsequent confusion/suggestions I would like to address it in a follow up PR if possible, since this PR is getting large and already starting to cut off comments due to length.